pkg/controllers/user/rbac/namespace_handler.go

```diff
@@ -39,12 +40,19 @@ type nsLifecycle struct {
 }
 
 func (n *nsLifecycle) Create(obj *v1.Namespace) (*v1.Namespace, error) {
+	resourceQuotaInit, err := IsNamespaceConditionSet(obj, ResourceQuotaInitCondition, true)
+	if err != nil {
+		return obj, err
+	}
+	if !resourceQuotaInit {
+		return obj, fmt.Errorf("resource quota is not set on the namespace [%s] yet; can't assign rbac roles", obj.Name)
```

Comment:
Not really an error, right? This will ultimately get logged as an error which will cause problems for users that monitor errors.
 
Can you use backoff logic to reschedule this instead, like [this](https://github.com/rancher/rancher/blob/f0b1f14a7cec8be876bfb62ff7ce824649868882/pkg/controllers/management/clusterprovisioner/provisioner.go#L190)*:
```
if p.backoff.IsInBackOffSinceUpdate(cluster.Name, time.Now()) {
		go func() {
			time.Sleep(p.backoff.Get(cluster.Name))
			p.ClusterController.Enqueue("", cluster.Name)
		}()
		return true, p.backoff.Get(cluster.Name)
	}
```


 \* I dont _really_ know if this code does what you need, but seems to fit the bill

Reply:
@cjellick not really an error, but I wanted to throw it so the framework does retry. I'll check the backoff logic